onebusaway-android/src/main/java/org/onebusaway/android/ui/HomeActivity.java

```diff
@@ -1237,6 +1293,22 @@ public void onClick(View arg0) {
         }
     }
 
+    // Check user profile restrictions on social features
+    private void checkSocialEnabled() {
+        if (Build.VERSION.SDK_INT > 18) {
+            Bundle restrictionsBundle = ((UserManager)getSystemService(Context.USER_SERVICE))
```

Comment:
Could you please refactor the entire code block that checks if `restrictionsBundle.containsKey(EMBEDDED_SOCIAL_KEY)` to a utility method in the `util` package with the name `SocialUtils.isSocialRestricted()``?  It should just return a boolean value, and you can then call this elsewhere in the app.  You should be able to use the Application context for getting the system service.